brickstrap-components.sh

```diff
@@ -0,0 +1,215 @@
+#!/bin/bash
+#
+# This file is part of brickstrap.
+#
+# brickstrap - create a foreign architecture rootfs using kernel namespaces,
+#              multistrap, and qemu usermode emulation and create a disk image
+#              using libguestfs
+#
+# Copyright (C) 2016 Johan Ouwerkerk <jm.ouwerkerk@gmail.com>
+#
+
+#
+# Note: this file is not meant to be executed, source it as a library of
+# functions instead. Variables used by the functions (other than stack) are
+# namespaced using the 'BR_' or 'BRP_' prefixes. Function names are namespaced
+# similarly, using the 'br_' and 'brp_' prefixes. See docs/namespacing.md for
+# more information on namespacing in brickstrap.
+#
+
+function brp_validate_project_name()
+{
+    [ -n "$BR_PROJECT" ] || fail "Project name must not be empty"
+    if [ -r "$BR_PROJECT" -a -d "$BR_PROJECT" ]; then
+        BR_PROJECT_DIR=$(readlink -f "$BR_PROJECT")
+    elif [ -r "$(br_script_path)/$BR_PROJECT" ] && \
+        [ -d "$(br_script_path)/$BR_PROJECT" ]; then
+        BR_PROJECT_DIR=$(readlink -f "$(br_script_path)/$BR_PROJECT")
+    else
+        fail "Invalid project name: '$BR_PROJECT'"
```

Comment:
I would rather have the error message say something to the effect that it could not find a matching project source folder. If you want to get fancy, you could even check that the folder contains (some of) the required files.